ci-operator/step-registry/upi/deprovision/azure/upi-deprovision-azure-commands.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set +e
+echo "Deprovisioning cluster ..."
+if [[ -f ${SHARED_DIR}/metadata.json ]]; then
+    INFRA_ID="$(jq -r .infraID ${SHARED_DIR}/metadata.json)"
+    AZURE_AUTH_CLIENT_ID=$(cat ${SHARED_DIR}/AZURE_AUTH_CLIENT_ID)
+    AZURE_AUTH_CLIENT_SECRET=$(cat ${SHARED_DIR}/AZURE_AUTH_CLIENT_SECRET)
+    AZURE_AUTH_TENANT_ID=$(cat ${SHARED_DIR}/AZURE_AUTH_TENANT_ID)
+    az login --service-principal -u $AZURE_AUTH_CLIENT_ID -p "$AZURE_AUTH_CLIENT_SECRET" --tenant $AZURE_AUTH_TENANT_ID
```

Comment:
Are we missing a call to `az account set --subscription $SUB_ID`? That seems necessary to distinguish which CI subscription we are using for this job?